src/ui/state/stakeholder.rs

```diff
@@ -526,6 +370,27 @@ impl StakeholderDelegateFundsState {
         Command::none()
     }
 
+    pub fn on_vault_acknowledge(&mut self, outpoint: String) -> Command<Message> {
+        if let Some(selected) = &mut self.selected_vault {
+            if selected.vault.outpoint() == outpoint {
+                return selected.update(self.revaultd.clone(), VaultMessage::Acknowledge(outpoint));
+            }
+        }
```

Comment:
You can probably remove this piece of code as this case is already covered below, isn't it?

Reply:
No really, the message is passed to the selected vault if a selected vault exists (1st part) else it selects the vault and passed the message (2 part of the code)

Reply:
I don't see why you can't use only the second part and eventually "reselect" the same vault, but fine, it's probably more efficient anyway :)